### PR TITLE
Set a new category if one doesn't exist.

### PR DESCRIPTION
**Pull request by tonypartridge**
Pull Request for Issue #15151. @brianteeman 

Just sets the original uncategorized category as the default selected. If you save, category is created. You can type a new category to create to proceed article creation.

Apply the patch, you can now create an article. IF not selecting a category it is placed in Uncategorized. 

Otherwise you can type a new category name and hit return, then save article and the new category and article is created.

### CODE REVIEW

**Comment by zero-24** [comment]
Can you take a look into drone?

http://213.160.72.75/joomla/joomla-cms/4916

**Comment by tonypartridge** [comment]
All done, drone should pass now.

File: administrator/components/com_categories/models/fields/categoryedit.php [hunk 1]
```diff
@@ -415,6 +415,16 @@ protected function getInput()
 		else
 			// Create a regular list.
 		{
+			if (count($options) === 0)
+			{
+				// All Categories have been deleted, so we need a new delete (This will create on save if selected)
```
Comment (brianteeman) [inline administrator/components/com_categories/models/fields/categoryedit.php]: need a new delete?

Comment (tonypartridge) [inline administrator/components/com_categories/models/fields/categoryedit.php]: Darn it! I had fixed that but lost it in a drone pr lol fixing now.

File: administrator/components/com_categories/models/fields/categoryedit.php [hunk 2]
```diff
@@ -415,6 +415,16 @@ protected function getInput()
...
+				$options[0]            = new stdClass;
+				$options[0]->value     = 'Uncategorised';
+				$options[0]->text      = 'uncategorised';
```
Comment (brianteeman) [inline administrator/components/com_categories/models/fields/categoryedit.php]: This should be Uncategorised as well

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 8a6e9ebb59f85714a7fa0df8e65dffc58ca4950f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16323">issues.joomla.org/tracker/joomla-cms/16323</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 8a6e9ebb59f85714a7fa0df8e65dffc58ca4950f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16323">issues.joomla.org/tracker/joomla-cms/16323</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Comment by tonypartridge** [comment]
Thanks guys!

**Comment by tonypartridge** [comment]
Can we get travis & AppVeyor rerun please?

**Comment by rdeutz** [comment]
set Milestone to 3.8 seems to me a new feature, so the 3.8 Release Lead should decide @mbabker

**Comment by brianteeman** [comment]
It's not really a new feature. It is fixing a bug as described in the original report #15151

**Comment by rdeutz** [comment]
@brianteeman so it was a new feature for 3.7.[0|1|2] ?

**Comment by tonypartridge** [comment]
Creating categories from category select was yes Robert in 3.7.

**Comment by rdeutz** [comment]
ok, than back to 3.7.3

**Comment by tonypartridge** [comment]
Thanks!

**Pull request merged**